src/logic/function/basic.lean

```diff
@@ -213,6 +213,23 @@ theorem cantor_injective {α : Type*} (f : (set α) → α) :
 cantor_surjective (λ a b, ∀ U, a = f U → U b) $
 right_inverse.surjective (λ U, funext $ λ a, propext ⟨λ h, h U rfl, λ h' U' e, i e ▸ h'⟩)
 
+/-- There is no surjection from `α : Type u` into `Type u`. This theorem
+  demonstrates why `Type : Type` would be inconsistent in Lean. -/
+theorem not_surjective_Type {α : Type u} (f : α → Type (max u v)) :
```

Comment:
If you were willing to move the theorem to `logic/small.lean`, then the proof would just be
```lean
λ hf, not_small_type $ small_of_surjective hf
```

Reply:
That file has a lot of dependencies, comparatively speaking. Perhaps the proof can be used to shorten this one, but I think it is fine where it is.